lru/lru.go

```diff
@@ -0,0 +1,100 @@
+package lru
+
+import "container/list"
+import "sync"
+
+// Cache represents a generic LRU cache with a very simple interface.
+type Cache interface {
+	// Put stores the given (key,value) pair, replacing existing value if key
+	// already exists.
+	Put(key, value interface{})
```

Comment:
Operations should be able to fail, hence they should all return an `error`. This allows future implementations to do things like I/O which may fail.